hydradx/tests/test_stableswap.py

```diff
@@ -219,33 +221,69 @@ def test_remove_asset(initial_pool: StableSwapPoolState):
         raise AssertionError("Asset values don't match.")
 
 
-@given(stableswap_config(precision=0.000000001))
-def test_buy_shares(initial_pool: StableSwapPoolState):
-    initial_agent = Agent(
-        holdings={tkn: 0 for tkn in initial_pool.asset_list + [initial_pool.unique_id]}
-    )
+@given(liq = st.lists(st.floats(min_value=100000, max_value=1000000), min_size=3, max_size=3),
+       amp = st.floats(min_value=5, max_value=1000),
+       pegs = st.lists(st.floats(min_value=0.1, max_value=10), min_size=2, max_size=2))
+def test_buy_shares_with_add_liquidity(liq: list[float], amp: float, pegs: list[float]):
+    tokens = {"A": liq[0], "B": liq[1], "C": liq[2]}
+    initial_pool = StableSwapPoolState(tokens, mpf(amp), trade_fee=mpf(0), peg=pegs)
+    initial_agent = Agent(holdings={tkn: 0 for tkn in initial_pool.asset_list + [initial_pool.unique_id]})
     # agent holds all the shares
     tkn_add = initial_pool.asset_list[0]
     pool_name = initial_pool.unique_id
     delta_tkn = 10
-    initial_agent.holdings.update({tkn_add: 10})
+    initial_agent.holdings.update({tkn_add: 2 * delta_tkn})
 
     add_liquidity_pool, add_liquidity_agent = stableswap.simulate_add_liquidity(
         initial_pool, initial_agent, delta_tkn, tkn_add
     )
     delta_shares = add_liquidity_agent.holdings[pool_name] - initial_agent.holdings[pool_name]
     buy_shares_pool, buy_shares_agent = stableswap.simulate_buy_shares(
-        initial_pool.copy(), initial_agent.copy(), delta_shares, tkn_add, fail_overdraft=False
+        initial_pool.copy(), initial_agent.copy(), delta_shares, tkn_add, fail_overdraft=True
```

Comment:
I suggest we just get rid of fail_overdraft. I don't think we do/should need it.